### DISCUSSION TITLE
Effect of not calling `disable_logging`/Why calling it causes error code -11

### DISCUSSION THREAD

**Discussion by ilan-gold** [rapidsai/rmm#1556] [Development]
Hello,

I have a very bizarre thing I'm seeing.  We are trying to run something with `asv` for benchmarking.   We have [a PR](https://github.com/scverse/rapids_singlecell/pull/176/files) for reference if you want to look.  We pulled the memory logger from the asv-[rapids benchmark hosted here](https://github.com/rapidsai/benchmark/tree/main/rapids_pytest_benchmark) but when we call `disable_logging` [here](https://github.com/scverse/rapids_singlecell/pull/176/files#diff-f5f3918ea461b367ebfa4bca0592c40e7fd8bc074e4f8a3c373dabc7a32db0c4R38) it causes exactly one of our benchmarks to fail with error code -11 and no thrown error or anything.  Deleting the line makes the benchmark pass.

So a workaround for us being deleting that line, what is the effect of not calling it?  Is there anything I should be worried about? `enable_logging` will be called for each benchmark with a different name each time.

Also any insight into _why_ this is happening would also be rad!

**Comment by harrism** [comment]
error 11 generally means a seg fault. Can you step into the `rmm.disable_logging()` call with a debugger to see where it fails internally?

`rmm.disable_logging` is not something I've worked on myself, but looking at the code it basically unwraps any per-device memory resource that is wrapped in a LoggingResourceAdapter. It could seg fault, I guess, if the wrapped memory resource was deleted somehow before that point. Does your code set the current device resource at all (from either Python or C++)?

**Comment by harrism** [comment]
If you are able to provide a simple, standalone reproducer then feel free to open an issue.
